src/Data/Nat/Divisibility.agda

```diff
@@ -25,24 +25,68 @@ open import Relation.Binary.Definitions
   using (Reflexive; Transitive; Antisymmetric; Decidable)
 import Relation.Binary.Reasoning.Preorder as PreorderReasoning
 open import Relation.Binary.PropositionalEquality.Core
-  using (_≡_; _≢_; refl; sym; trans; cong; cong₂; subst)
+  using (_≡_; _≢_; refl; sym; cong; subst)
 open import Relation.Binary.Reasoning.Syntax
-import Relation.Binary.PropositionalEquality.Properties as PropEq
+open import Relation.Binary.PropositionalEquality.Properties
+  using (isEquivalence; module ≡-Reasoning)
+
+private
+  variable d m n o p : ℕ
+
+
+------------------------------------------------------------------------
+-- Definition and derived properties
+
+open import Data.Nat.Divisibility.Core public hiding (*-pres-∣)
+
+module _ (m∣n : m ∣ n) where
+
+  open _∣_ m∣n
+  open ≤-Reasoning
+
+  instance
+    quotient≢0 : .{{NonZero n}} → NonZero quotient
+    quotient≢0 rewrite equality = m*n≢0⇒m≢0 quotient
```

Comment:
Making this an instance is pointless as it has an explicit argument (which would be clearer if the explicit parameter wasn't lifted up to anonymous module)

Reply:
I think in earlier versions of this PR, we went round in circles about making this an instance or not. Previously, when it was manifest field, I think you argued for it being an instance (against Nathan); when I moved it up to a lemma in the anonymous module, I didn't rethink the question. Will remove the `instance` ...